mediator/med_methods_mod.F90

```diff
@@ -2506,12 +2506,14 @@ subroutine med_methods_FB_getmesh(FB, mesh, rc)
   end subroutine med_methods_FB_getmesh
 
   !-----------------------------------------------------------------------------
-  subroutine med_methods_FB_check_for_nans(FB, rc)
-
-    use ESMF, only : ESMF_FieldBundle, ESMF_Field, ESMF_FieldBundleGet, ESMF_FieldGet
-
+  subroutine med_methods_FB_check_for_nans(gcomp, FB, maintask, logunit, rc)
```

Comment:
I don't understand why you don't set checkfornans as a module variable - and introduce another module variable that is determine_checkfornans. If determine_checkfornans is .false. - then set checkfornans. I would think that would be cleaner than adding a save attribute to the logicals (firstcall and checkfornans).